src/emit/expression.rs

```diff
@@ -1919,3 +1941,49 @@ fn runtime_cast<'a>(
         val
     }
 }
+
+fn string_to_basic_value<'a>(
+    bin: &Binary<'a>,
+    ns: &Namespace,
+    input: String,
+) -> BasicValueEnum<'a> {
+    let elem = Type::Bytes(1);
+    let size = bin.context.i32_type().const_int(input.len() as u64, false);
+
+    let elem_size = bin
+        .llvm_type(&elem, ns)
+        .size_of()
+        .unwrap()
+        .const_cast(bin.context.i32_type(), false);
+
+    let init = &Option::Some(input.as_bytes().to_vec());
+    bin.vector_new(size, elem_size, init.as_ref()).into()
+}
+
+pub(super) fn print_runtime_error<'a>(
+    bin: &Binary<'a>,
+    ns: &Namespace,
+    error: &String,
+    loc: Option<Loc>,
+) -> BasicValueEnum<'a> {
+    let error_with_line = if let Some(loc) = loc {
+        if let Loc::Codegen = loc {
+            format!("{error} in  codegen")
+        } else {
+            let file_no = loc.file_no();
```

Comment:
This doesn't cover all cases, see `UniswapV2Pair.sol` for an example where the location is neither `Codegen` nor `File`, which crashes the compiler.